app/controllers/runtime/organizations_controller.rb

```diff
@@ -59,14 +59,44 @@ def self.translate_validation_exception(e, attributes)
       end
     end
 
-    def before_update(obj)
+    def before_update(org)
       if request_attrs['default_isolation_segment_guid']
         raise CloudController::Errors::ApiError.new_from_details(
           'ResourceNotFound',
           'Could not find Isolation Segment to set as the default.') unless IsolationSegmentModel.first(guid: request_attrs['default_isolation_segment_guid'])
       end
 
-      super(obj)
+
+
+      super(org)
+    end
+
+    def update(guid)
```

Comment:
@ericpromislow Unfortunately, it's not that simple - b/c of the way API dispatch happens, if you cross lines between objects it gets a lot hairier.  The ideal way to do this would be to move all of the role assignment (not just these update methods) into workflow actions, but that's a pretty sizable refactor.

If you look, all this is is the `update()` method copied from the `ModelController`  - which I had to do b/c otherwise it's instantiating a new controller object on each`before_update()` and `after_update()` call - which breaks the logic, b/c you have to have a single controller object to hold the Tx state as it sits right now.  So no, I don't think this can be moved into a library method (nor do I think it should be), b/c I think that ends up just being throwaway code.